src/os/filestore/FileStore.cc

```diff
@@ -237,18 +237,22 @@ int FileStore::lfn_open(const coll_t& cid,
 			const ghobject_t& oid,
 			bool create,
 			FDRef *outfd,
-                        Index *index)
+			Index *index,
+			bool directio)
 {
   assert(outfd);
   int r = 0;
   bool need_lock = true;
   int flags = O_RDWR;
+  FDCache *cachep = directio ? &fdcache_direct : &fdcache;
 
   if (create)
     flags |= O_CREAT;
   if (g_conf->filestore_odsync_write) {
     flags |= O_DSYNC;
   }
+  if (directio)
+    flags |= O_DIRECT;
```

Comment:
SSD writes are indeed limited by the number of erase cycles, but there is no limit to the number of reads that I'm aware of. This feature only applies to reads. The suggested use is only for high-performance storage such as SSDs and newer memory technologies that are coming up. For rotating media this wouldn't make sense, as you note, and so it is disabled by default.